bundles/org.openhab.binding.mercedesme/src/main/java/org/openhab/binding/mercedesme/internal/api/Authorization.java

```diff
@@ -0,0 +1,533 @@
+                resumeResponse.getRequest().getURI());
+        if (resumeResponse.getStatus() == 200) {
+            String response = resumeResponse.getRequest().getURI().getQuery();
+            Map<String, String> params = Utils.getQueryParams(response);
+            resumeUrl = params.get("resume");
+        }
+        if (resumeUrl != null) {
+            return resumeUrl;
+        } else {
+            throw new MercedesMeAuthException("Failed to get resume URL. HTTP " + resumeResponse.getStatus());
+        }
+    }
+
+    /**
+     * Send user agent.
+     *
+     * @throws MercedesMeAuthException if response status isn't correct
+     * @throws MercedesMeApiException if an error occurs during API call
+     */
+    private void sendUserAgent(HttpClient loginHttpClient) throws MercedesMeAuthException, MercedesMeApiException {
+        Request agentRequest = loginHttpClient.POST(baseUrl + "/ciam/auth/ua");
+        agentRequest.header(HttpHeader.USER_AGENT, Constants.AUTH_USER_AGENT);
+        agentRequest.header(HttpHeader.ACCEPT_LANGUAGE, Constants.AUTH_LANGUAGE);
+        agentRequest.header(HttpHeader.ACCEPT, "*/*");
+        agentRequest.header(HttpHeader.CONTENT_TYPE, CONTENT_TYPE_JSON);
+        agentRequest.header(HttpHeader.ORIGIN, baseUrl);
+
+        JSONObject agentContent = new JSONObject();
+        agentContent.put("browserName", "Mobile Safari");
+        agentContent.put("browserVersion", "15.6.6");
+        agentContent.put("osName", "iOS");
+        agentRequest.content(new StringContentProvider(agentContent.toString(), "utf-8"));
+
+        ContentResponse agentResponse = send(agentRequest);
+        logger.trace("Step 2: Post Agent {} - {}", agentResponse.getStatus(), agentResponse.getContentAsString());
+        if (agentResponse.getStatus() != 200) {
+            throw new MercedesMeAuthException("Failed to post user agent. HTTP " + agentResponse.getStatus());
+        }
+    }
+
+    /**
+     * Send user name.
+     *
+     * @throws MercedesMeAuthException if response status isn't correct
+     * @throws MercedesMeApiException if an error occurs during API call
+     */
+    private void sendUsername(HttpClient loginHttpClient) throws MercedesMeAuthException, MercedesMeApiException {
+        Request userRequest = loginHttpClient.POST(baseUrl + "/ciam/auth/login/user");
+        userRequest.header(HttpHeader.USER_AGENT, Constants.AUTH_USER_AGENT);
+        userRequest.header(HttpHeader.ACCEPT_LANGUAGE, Constants.AUTH_LANGUAGE);
+        userRequest.header(HttpHeader.ACCEPT, CONTENT_TYPE_JSON + ", text/plain, */*");
+        userRequest.header(HttpHeader.CONTENT_TYPE, CONTENT_TYPE_JSON);
+        userRequest.header(HttpHeader.ORIGIN, baseUrl);
+        userRequest.header(HttpHeader.REFERER, baseUrl + "/ciam/auth/login");
+
+        JSONObject userContent = new JSONObject();
+        userContent.put("username", config.email);
+        userRequest.content(new StringContentProvider(userContent.toString(), "utf-8"));
```

Comment:
See comment further down.
```suggestion
        userRequest.agent(Constants.AUTH_USER_AGENT);
        userRequest.header(HttpHeader.ACCEPT_LANGUAGE, Constants.AUTH_LANGUAGE);
        userRequest.header(HttpHeader.ACCEPT, CONTENT_TYPE_JSON + ", text/plain, */*");
        userRequest.header(HttpHeader.ORIGIN, baseUrl);
        userRequest.header(HttpHeader.REFERER, baseUrl + "/ciam/auth/login");

        JSONObject userContent = new JSONObject();
        userContent.put("username", config.email);
        userRequest.content(new StringContentProvider(userContent.toString(), "utf-8"), CONTENT_TYPE_JSON);
```